pkg/util/intern/intern.go

```diff
@@ -0,0 +1,70 @@
+package intern
+
+import (
+	"sync"
+	"unsafe"
+
+	pq "github.com/parquet-go/parquet-go"
+)
+
+type Interner struct {
+	mtx sync.RWMutex
+	m   map[string][]byte
+}
+
+func New() *Interner {
+	return &Interner{m: make(map[string][]byte)}
+}
+
+func (i *Interner) UnsafeClone(v *pq.Value) pq.Value {
+	a := *(*aValue)(unsafe.Pointer(v))
```

Comment:
this works b/c the pq.Value struct just happens to line up with the aValue struct :)? definitely need some testing on that one

also don't need to do this unless we're in the first switch case.

Reply:
Yes :) It's a bit unsafe, but I didn't find another way of accessing those fields.

I've added a test (`Test_pqValue`) for converting the value back and forth and checking integrity. That should cover your concerns.